lib/matobj.gi

```diff
@@ -1380,6 +1380,38 @@ InstallMethod( InverseSameMutability,
     fi;
     end );
 
+InstallMethod( IsZero,
+    [ IsRowListMatrix and IsMatrixObj ],
+    function( mat )
+    local ncols, row;
+
+    ncols:= NrCols( mat );
+    for row in mat do
+      if PositionNonZero( row ) <= ncols then
```

Comment:
I was going to suggest that this might be clearer if it was:
```suggestion
      if PositionNonZero( row ) <> fail then
```
only to discover that `PositionNonZero` returns `Length(row) + 1` if no non-zero is located. Is this intentional, it seems to be at odds with `Position`.

Reply:
The documentation says that `PositionNonZero` is a special case of `PositionNot`, and `PositionNot` treats unbound entries as "not the element which shall be avoided".